### DISCUSSION TITLE
Saving in-memory sqlite3 database to a file/bytestream?

### DISCUSSION THREAD

**Discussion by arashster** [jeremyevans/sequel#2022] [General]
Is there a way to convert a Sequel in-memory sqlite3 database to a bytestream?
For example, I've created an in-memory sqlite3 db, but now want to convert it to binary representation after the fact.

Is this possible?

Similarly, is it possible to load that bytestream after the fact as an in-memory sqlite3 database?

What I want to do is avoid having to read/write to a file, and do everything in-memory but have the ability to checkpoint/save the database if need be to a file (or another file-like location like an S3 bucket).

**Comment by jeremyevans** [comment]
This would be a question for the SQLite developers, not for Sequel.

**Comment by arashster** [comment]
Thanks for the quick response!
That makes sense.. 
Thinking out loud, but was hoping there might be a way to implement a simple abstractions like 
`DB.dump` or` DB.load(str)` specifically for sqlite3
